pkg/kritis/cryptolib/pgp_test.go

```diff
@@ -136,3 +304,100 @@ func TestVerifyPgp(t *testing.T) {
 		})
 	}
 }
+
+func TestCreateAttestation_WellFormedAttestation(t *testing.T) {
+	tcs := []struct {
+		name               string
+		privateKey         []byte
+		expectNewSignerErr bool
+	}{
+		{
+			name:               "correct signature over payload",
+			privateKey:         []byte(gpgPrivateKey),
+			expectNewSignerErr: false,
+		},
+		{
+			name:               "invalid private key",
+			privateKey:         []byte("invalid-private-key"),
+			expectNewSignerErr: true,
+		},
+	}
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			signer, err := NewPgpSigner(tc.privateKey)
+			if tc.expectNewSignerErr {
+				if err == nil {
+					t.Fatalf("Expected error but returned none")
+				}
+			} else {
+				if err != nil {
+					t.Fatalf("Unexpected error: %v", err)
+				}
+				att, err := signer.CreateAttestation([]byte(payload))
+				if err != nil {
+					t.Fatalf("Error creating the Attestation: %v", err)
+				}
+
+				// Check attestation contents
+				if att.SerializedPayload != nil {
+					t.Errorf("SerializedPayload does not match expected: got: %v, want: %v", att.SerializedPayload, nil)
+				}
+				if att.PublicKeyID != gpgPublicKeyID {
+					t.Errorf("PublicKeyID does not match expected: got: %q, want %q", att.PublicKeyID, gpgPublicKeyID)
+				}
+			}
+		})
+	}
+}
+
+// Contents of generated signatures depend on external factors, including time
```

Comment:
// Test create attestation by asserting the created attestation can be verified. We don't compare the created attestation to a hard-coded expected result because the contents of generated signatures depend on external factors, including time  and date.

Reply:
Done.